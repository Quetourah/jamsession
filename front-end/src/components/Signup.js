
import React from 'react';


import 'bootstrap/dist/css/bootstrap.min.css';
import {Form,InputGroup,Button,Col} from 'react-bootstrap';

function Signup() {
    
  
    return (
      <Form >
        <Form.Row>
          <Form.Group as={Col} md="4" >
            <Form.Label>First name</Form.Label>
            <Form.Control
              required
              type="text"
              placeholder="First name"
              defaultValue="Mark"
            />
            
          </Form.Group>
          <Form.Group as={Col} md="4" >
            <Form.Label>Last name</Form.Label>
            <Form.Control
              required
              type="text"
              placeholder="Last name"
              defaultValue="Otto"
            />
            
          </Form.Group>
          <Form.Group as={Col} md="4" >
            <Form.Label>Username</Form.Label>
            <InputGroup>
              <InputGroup.Prepend>
                <InputGroup.Text id="inputGroupPrepend">@</InputGroup.Text>
              </InputGroup.Prepend>
              <Form.Control
                type="text"
                placeholder="Username"
                aria-describedby="inputGroupPrepend"
                required
              />
              <Form.Control.Feedback type="invalid">
                Please choose a username.
              </Form.Control.Feedback>
            </InputGroup>
          </Form.Group>
        
          <Form.Group as={Col} md="6" >
            <Form.Label>City</Form.Label>
            <Form.Control type="text" placeholder="City" required />
            
          </Form.Group>
          <Form.Group as={Col} md="3" >
            <Form.Label>State</Form.Label>
            <Form.Control type="text" placeholder="State" required />
            
          </Form.Group>
          <Form.Group as={Col} md="3" >
            <Form.Label>Zip</Form.Label>
            <Form.Control type="text" placeholder="Zip" required />
            
          </Form.Group>
        </Form.Row>
        <Form.Group>
          <Form.Check
            required
            label="Agree to terms and conditions"
            feedback="You must agree before submitting."
          />
        </Form.Group>
        <Button type="submit">SignUp</Button>
      </Form>
    );
  }
  
export default Signup;