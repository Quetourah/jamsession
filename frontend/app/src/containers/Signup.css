@media all and (min-width: 480px) {
  .Signup {
    background-image: url("https://images.unsplash.com/photo-1510279410431-2d0808d69bf8?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1650&q=80");
  }
    .Signup{
      padding: 60px 0;
    }
  
    .Signup form {
      margin: 0 auto;
      max-width: 320px;
      color: red;
    }
  }