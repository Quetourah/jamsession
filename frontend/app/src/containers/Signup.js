import React, { Component } from "react";
import {HelpBlock,FormGroup,FormControl,ControlLabel} from "react-bootstrap";
import { Auth } from "aws-amplify";
import LoaderButton from "../components/LoaderButton";
import "./Signup.css";

export default class Signup extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isLoading: false,
      email: "",
      username: "",
      password: "",
      confirmPassword: "",
      confirmationCode: "",
      newUser: null
    };
  }

  validateForm() {
    return (
      this.state.email.length > 0 &&
      this.state.username.length>0 &&
      this.state.password.length > 0 &&
      this.state.password === this.state.confirmPassword
    );
  }

  validateConfirmationForm() {
    return this.state.confirmationCode.length > 0;
  }

  handleChange = event => {
    this.setState({
      [event.target.id]: event.target.value
    });
  }

  handleSubmit = async event => {
    event.preventDefault();

    this.setState({ isLoading: true });

    try {
      const newUser = await Auth.signUp({
        username: this.state.username,
        password: this.state.password, 
        attributes: {email: this.state.email}
      });
      this.setState({
        newUser
      });
    } catch (e) {
      alert(e.message);
    }

    this.setState({ isLoading: false });
  }

  handleConfirmationSubmit = async event => {
    event.preventDefault();

    this.setState({ isLoading: true });

    try {
      await Auth.confirmSignUp(this.state.email, this.state.username, this.state.confirmationCode);
      await Auth.signIn(this.state.username, this.state.email, this.state.password);

      this.props.userHasAuthenticated(true);
      this.props.history.push("/");
    } catch (e) {
      alert(e.message);
      this.setState({ isLoading: false });
    }
  }

  renderConfirmationForm() {
    return (
      <form onSubmit={this.handleConfirmationSubmit}>
        <FormGroup controlId="confirmationCode" bsSize="large">
          <ControlLabel>Confirmation Code</ControlLabel>
          <FormControl
            autoFocus
            type="tel"
            value={this.state.confirmationCode}
            onChange={this.handleChange}
          />
          <HelpBlock>Please check your email for the code.</HelpBlock>
        </FormGroup>
        <LoaderButton
          block
          bsSize="large"
          disabled={!this.validateConfirmationForm()}
          type="submit"
          isLoading={this.state.isLoading}
          text="Verify"
          loadingText="Verifying…"
        />
      </form>
    );
  }

  renderForm() {
    return (
      <form onSubmit={this.handleSubmit}>
        <FormGroup controlId="email" bsSize="large">
          <ControlLabel>Email</ControlLabel>
          <FormControl
            autoFocus
            type="email"
            value={this.state.email}
            onChange={this.handleChange}
          />
        </FormGroup>
        <FormGroup controlId="username" bsSize="large">
          <ControlLabel>Username</ControlLabel>
          <FormControl
            autoFocus
            type="text"
            value={this.state.username}
            onChange={this.handleChange}
          />
        </FormGroup>
        <FormGroup controlId="password" bsSize="large">
          <ControlLabel>Password</ControlLabel>
          <FormControl
            value={this.state.password}
            onChange={this.handleChange}
            type="password"
          />
        </FormGroup>
        <HelpBlock>Make sure password has at least 8 characters. Include upper case letters, numbers and special characters.</HelpBlock>
        <FormGroup controlId="confirmPassword" bsSize="large">
          <ControlLabel>Confirm Password</ControlLabel>
          <FormControl
            value={this.state.confirmPassword}
            onChange={this.handleChange}
            type="password"
          />
        </FormGroup>
        <LoaderButton
          block
          bsSize="large"
          disabled={!this.validateForm()}
          type="submit"
          isLoading={this.state.isLoading}
          text="Sign Up"
          loadingText="Signing up…"
        />
      </form>
    );
  }

  render() {
    return (
      <div className="Signup">
        {this.state.newUser === null
          ? this.renderForm()
          : this.renderConfirmationForm()}
      </div>
    );
  }
}