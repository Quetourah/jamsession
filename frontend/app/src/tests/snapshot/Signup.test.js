import React from 'react';
import Signup from '../../containers/Signup';
import renderer from 'react-test-renderer';

it('renders sign up page correctly', () => {
  const tree = renderer
    .create(<Signup/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

