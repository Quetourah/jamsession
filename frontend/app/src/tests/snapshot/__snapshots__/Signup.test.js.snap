// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders sign up page correctly 1`] = `
<div
  className="Signup"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group form-group-lg"
    >
      <label
        className="control-label"
        htmlFor="email"
      >
        Email
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group form-group-lg"
    >
      <label
        className="control-label"
        htmlFor="username"
      >
        Username
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="username"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group form-group-lg"
    >
      <label
        className="control-label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-control"
        id="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <span
      className="help-block"
    >
      Make sure password has at least 8 characters. Include upper case letters, numbers and special characters.
    </span>
    <div
      className="form-group form-group-lg"
    >
      <label
        className="control-label"
        htmlFor="confirmPassword"
      >
        Confirm Password
      </label>
      <input
        className="form-control"
        id="confirmPassword"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <button
      className="LoaderButton  btn btn-lg btn-default btn-block"
      disabled={true}
      type="submit"
    >
      Sign Up
    </button>
  </form>
</div>
`;
